From 29d998206642ff27fd1717b189fe012efd96e700 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marcin=20Wcis=C5=82o?= <marcin.wcislo@conclusive.pl>
Date: Tue, 4 Nov 2025 20:32:47 +0100
Subject: [PATCH] netfilter: nf_tables: Add null check for
 nla_nest_start_noflag() in nft_dump_basechain_hook()
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

jira VULN-430
subsystem-update netfilter: centos-stream-9 e4726e3376a4798e40b1bce02252a209fcf0de7c
commit-author Gavrilov Ilia <Ilia.Gavrilov@infotecs.ru>
commit bd058763a624a1fb5c20f3c46e632d623c043676
upstream-diff Used the cleanly applying 9.4 backport
  04580de505708123813c20c2046bfffd2fbf7b0d

The nla_nest_start_noflag() function may fail and return NULL;
the return value needs to be checked.

Found by InfoTeCS on behalf of Linux Verification Center
(linuxtesting.org) with SVACE.

Fixes: d54725cd11a5 ("netfilter: nf_tables: support for multiple devices per netdev hook")
	Signed-off-by: Gavrilov Ilia <Ilia.Gavrilov@infotecs.ru>
	Signed-off-by: Florian Westphal <fw@strlen.de>
	Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>
(cherry picked from commit bd058763a624a1fb5c20f3c46e632d623c043676)
	Signed-off-by: Marcin Wcisło <marcin.wcislo@conclusive.pl>
---
 net/netfilter/nf_tables_api.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/net/netfilter/nf_tables_api.c b/net/netfilter/nf_tables_api.c
index 1ae9e16055d6..dc17f48684fe 100644
--- a/net/netfilter/nf_tables_api.c
+++ b/net/netfilter/nf_tables_api.c
@@ -1600,6 +1600,10 @@ static int nft_dump_basechain_hook(struct sk_buff *skb, int family,
 
 	if (nft_base_chain_netdev(family, ops->hooknum)) {
 		nest_devs = nla_nest_start_noflag(skb, NFTA_HOOK_DEVS);
+
+		if (!nest_devs)
+			goto nla_put_failure;
+
 		list_for_each_entry(hook, &basechain->hook_list, list) {
 			if (!first)
 				first = hook;
-- 
2.47.3

